README.md

```diff
@@ -100,6 +99,85 @@ The output of the CSV Write component will be a message with an attachment.  In
 order to access this attachment, the component following the CSV Write must be
 able to handle file attachments.
 
+### Write CSV attachment from JSON
+
+* `Include Header` - this select configures output behavior of the component. If option is `Yes` or no value chosen than header of csv file will be written to attachment, this is default behavior. If value `No` selected than csv header will be omitted from attachment.
+* `Separator` - this select configures type of CSV delimiter in an output file. There are next options: `Comma (,)`, `Semicolon (;)`, `Space ( )`, `Tab (\t)`. 
+
+This action will combine multiple incoming events into a CSV file until there is a gap
+of more than 10 seconds between events. Afterwards, the CSV file will be closed
+and attached to the outgoing message.
+
+This action will convert an incoming array into a CSV file by following approach:
+
+* Header inherits names of keys from the input message;
+* Payload will store data from Values of relevant Keys (Columns).
+
+The keys of an input JSON will be published as the header in the first row. For each incoming
+event, the value for each header will be `stringified` and written as the value
+for that cell. All other properties will be ignored. For example, headers
+`foo,bar` along with the following JSON events:
+
+```
+{"foo":"myfoo", "bar":"mybar"}
+{"foo":"myfoo", "bar":[1,2]}
+{"bar":"mybar", "baz":"mybaz"}
+```
+
+will produce the following `.csv` file:
+
+```
+foo,bar
+myfoo,mybar
+myfoo,"[1,2]"
+,mybar
+```
+
+The output of the CSV Write component will be a message with an attachment.  In
+order to access this attachment, the component following the CSV Write must be
+able to handle file attachments.
+
+### Write CSV attachment from Array
+
+* `Include Header` - this select configures output behavior of the component. If option is `Yes` or no value chosen than header of csv file will be written to attachment, this is default behavior. If value `No` selected than csv header will be omitted from attachment.
+* `Separator` - this select configures type of CSV delimiter in an output file. There are next options: `Comma (,)`, `Semicolon (;)`, `Space ( )`, `Tab (\t)`. 
+
+This action will convert an incoming array into a CSV file by following approach:
+
+* Header inherits names of keys from the input message;
+* Payload will store data from Values of relevant Keys (Columns).
+
+Requirements:
+
+* The inbound message is an JSON Array of Object with identical structure;
+* Each JSON object has plain structure without nested levels. 
+
+The keys of an input JSON will be published as the header in the first row. For each incoming
+event, the value for each header will be `stringified` and written as the value
+for that cell. All other properties will be ignored. For example, headers
```

Comment:
> All other properties will be ignored.

What are other properties?

Reply:
Means, if the first object has keys: `Name`, `Surname` and the second - `Name`, `Surname`, `Email`, the `Email` property will be ignored and won't be added to an output file.
For example,
```
[
  {
    Name: 'John',
    Surname: 'Doe'
  },
  {
    Name: 'Jane',
    Surname: 'Doe',
    Email: 'jane.doe@email.domain'
  }
]
```
will be transformed into
```
Name,Surname
John,Doe
Jane,Doe
```

In opposite, if the first object has keys: `Name`, `Surname`, `Email` and the second - `Name`, `Surname`, all three properties will present in an output file.
```
[
  {
    Name: 'John',
    Surname: 'Doe',
    Email: 'john.doe@email.domain'
  },
  {
    Name: 'Jane',
    Surname: 'Doe'
  }
]
```
will be transformed into
```
Name,Surname,Emal
John,Doe,john.doe@email.domain
Jane,Doe,
```